lib/optimizely/event_builder.rb

```diff
@@ -69,35 +72,46 @@ def get_common_params(user_id, attributes)
         attribute_value = attributes[attribute_key]
         next if attribute_value.nil?
 
-        # Skip attributes not in the datafile
-        attribute_id = @config.get_attribute_id(attribute_key)
-        next unless attribute_id
-
-        feature = {
-          entity_id: attribute_id,
-          key: attribute_key,
-          type: CUSTOM_ATTRIBUTE_FEATURE_TYPE,
-          value: attribute_value
-        }
+        if attribute_key.eql? RESERVED_ATTRIBUTE_KEY_BUCKETING_ID
+          # TODO (Copied from PHP-SDK) (Alda): the type for bucketing ID attribute may change so 
+          # that custom attributes are not overloaded          
+          feature = {
+            entity_id: RESERVED_ATTRIBUTE_KEY_BUCKETING_ID,
+            key: RESERVED_ATTRIBUTE_KEY_BUCKETING_ID_EVENT_PARAM_KEY,
+            type: CUSTOM_ATTRIBUTE_FEATURE_TYPE,
+            value: attribute_value
+          }         
+        else
+          # Skip attributes not in the datafile
+          attribute_id = @config.get_attribute_id(attribute_key)
+          next unless attribute_id
+
+          feature = {
+            entity_id: attribute_id,
+            key: attribute_key,
+            type: CUSTOM_ATTRIBUTE_FEATURE_TYPE,
+            value: attribute_value
+          }
 
-        visitor_attributes.push(feature)
         end
-      end 
-
-      common_params = {
-        account_id: @config.account_id,
-        project_id: @config.project_id,
-        visitors: [
-          {
-            attributes: visitor_attributes,
-            snapshots: [],
-            visitor_id: user_id
-          }
-          ],
-          revision: @config.revision,
-          client_name: CLIENT_ENGINE,
-          client_version: VERSION
+        visitor_attributes.push(feature)
```

Comment:
please add a condition if feature is not nil. there is a chance attribute_id can be nil and feature object won't be initialized.

Reply:
Shouldn't be an issue. 
Line : next unless attribute_id functions as 'continue'. The iter will go to the top of the do block if addtribute_id is nil